settings/src/lib.rs

```diff
@@ -239,8 +242,10 @@ pub struct PaymentSettings {
     pub close_fraction: Int256,
     /// The amount of billing cycles a node can fall behind without being subjected to the threshold
     pub buffer_period: u32,
-    /// Our own eth address
-    pub eth_address: EthAddress,
+    /// Our own eth address, derived from eth_private_key and stored for ease of access
+    pub eth_address: Option<Address>,
```

Comment:
We don't need the address. It's another thing for the user to worry about. Any address is deterministically derived off the private key (last 20 bytes of the Keccak hash of the public key which in itself also has a deterministic relationship with the private key)

Reply:
I was on the fence about that. You can see in clu that on startup we check that address matches the private key and fix it if it doesn't. We could totally ignore storing it but in doing so we'd have to compute our own address quite a bit, I'm not really sure which way is better.